Can't reference `Self` type when using associated refinement

When I try to use an associated refinement in a function spec, Flux does not seem to like it when I pass in a `Self`.
For example, when I verify:

```Rust
#[flux_rs::refined_by(powered:bool)]
pub struct MPU<const MIN_REGION_SIZE: usize> {
    #[field(bool[powered])]
    powered: bool,
}

#[flux_rs::assoc(fn enabled(self: Self) -> bool )]
pub trait MyTrait {
    #[flux_rs::sig(fn(self: &strg Self) ensures self: Self{r: <Self as MyTrait>::enabled(r)})]
    fn enable_app_mpu(&mut self);
}

#[flux_rs::assoc(fn enabled(self: Self) -> bool {self.powered} )]
impl<const MIN_REGION_SIZE: usize> MyTrait for MPU<MIN_REGION_SIZE> {
    #[flux_rs::sig(fn(self: &strg Self) ensures self: Self{r: <Self as MyTrait>::enabled(r)})]
    fn enable_app_mpu(&mut self) {
        self.powered = true;
    }
}
```

I get the error:
```
error[E0999]: mismatched sorts
   --> src/main.rs:904:90
    |
904 |     #[flux_rs::sig(fn(self: &strg Self) ensures self: Self{r: <Self as MyTrait>::enabled(r)})]
    |                                                                                          ^ expected `{impl#0}::sort`, found `MPU`

```

It seems like Flux is expecting an instance of `MyTrait` rather than an `MPU` and does not recognize that MPU implements MyTrait. I'm pretty sure I'm just missing some syntax, but I could use a pointer lol.

On a related note, are there any plans for documentation on associated refinements? They are extremely useful, but a little hard to use due to the weird syntax.

--- a/tests/tests/pos/surface/issue-923.rs
+++ b/tests/tests/pos/surface/issue-923.rs
@@ -1,0 +1,19 @@
+#[flux_rs::refined_by(powered:bool)]
+pub struct MPU<const MIN_REGION_SIZE: usize> {
+    #[field(bool[powered])]
+    powered: bool,
+}
+
+#[flux_rs::assoc(fn enabled(self: Self) -> bool )]
+pub trait MyTrait {
+    #[flux_rs::sig(fn(self: &strg Self) ensures self: Self{r: <Self as MyTrait>::enabled(r)})]
+    fn enable_app_mpu(&mut self);
+}
+
+#[flux_rs::assoc(fn enabled(self: Self) -> bool {self.powered} )]
+impl<const MIN_REGION_SIZE: usize> MyTrait for MPU<MIN_REGION_SIZE> {
+    #[flux_rs::sig(fn(self: &strg Self) ensures self: Self{r: <Self as MyTrait>::enabled(r)})]
+    fn enable_app_mpu(&mut self) {
+        self.powered = true;
+    }
+}

